### PR TITLE
Add scroll capacities to menu

### PR DESCRIPTION
**Pull request by anxuae**
This is the full implementation of #15.

### CODE REVIEW

**Comment by ppizarror** [comment]
Tests failed on pygame 2.7 https://travis-ci.org/github/ppizarror/pygame-menu/jobs/672115273

**Comment by ppizarror** [comment]
Introduced `_calculate_column_widths` method. The column widths are calculated, before scrolling, by assigning the maximum column width by the factor of the column. If the total width was 600px, and the column factor was (1, 2, 1) then each column will have the (25%, 50%, 25%) of the width.

Because of the new scroll area format. This is valid only if `force_fit_text` is enabled. If that parameter is true the width will be fixed and no scroll is available.

So, as the scroll let's the user to dynamically expand the width of the surface, this method should calculate then the width of each column (as the max of each row per column) and `column factor` is now only valid for `force_fit_text`

**Comment by ppizarror** [comment]
Maybe, after the column width is calculated by the maximum of it's rows, it can be multiplied by `column_weigths`... It makes more sense to me

**Comment by ppizarror** [comment]
As I try to formalize, `column_weigths` is confusing to me. So I propose the following:

Add `column_widths' as a Menu parameter. If None then `column_widths` will be calculated automatically based. For example

`Menu(width=800, colummn_width=(100, None, None, 200))`

First and Last column will have 100px and 200px maximum. Column 3 and 4 will have a minimum width of `(800-100-200)/2 = 250`

Columns that does not have any maximum value (None in `column_widths`) will grow based on the widgets in them, as new widgets are appended to the Menu. For example

`column_width[1] = max(widget_widths in column)`

By default. If the widget width is greater than the column, and `force_fit_text` is enabled (Will be true by default) then the widget will be fitted. If the widget width is lower than the column, nothing happens.

@anxuae what you think about this new approach?

**Comment by anxuae** [comment]
It seem usable.
By default, `column_widths=None` would mean that each column as the width of the largest widget in it, right?

**Comment by ppizarror** [comment]
Yes!

**Comment by ppizarror** [comment]
Added fully column support. Also Added an option to center vertically the menus.

File: pygameMenu/scrollarea.py [hunk 1]
```diff
@@ -297,6 +297,20 @@ def get_view_rect(self):
 
         return rect
 
+    def get_actual_scrollbar_thickness(self, orientation):
```
Comment (anxuae) [inline pygameMenu/scrollarea.py:300]: naming `get_scrollbar_thickness` is enough , no?

Comment (ppizarror) [reply pygameMenu/scrollarea.py:300]: that's better

**Comment by ppizarror** [comment]
@anxuae. There's an inconsistency in `draw_region_y`.

After the y scrollbar returns to the first position. In the test I made (see gif) at first the margin is valid (added `draw_region_y=30`), But after returning to the first position the margin was deleted.

https://i.imgur.com/KXUOohR.gif

File: pygameMenu/scrollarea.py [hunk 2]
```diff
@@ -297,6 +297,20 @@ def get_view_rect(self):
...
+        """
+        Return the scroll thickness of the area. If it's hidden return zero.
+
+        :param orientation: Orientation of the scroll
+        :type orientation: basestring
+        :return: Thickness in px
+        :rtype: int
+        """
+        if orientation == _locals.ORIENTATION_HORIZONTAL and self._world.get_width() > self._rect.width or \
```
Comment (anxuae) [inline pygameMenu/scrollarea.py:309]: The condition `self._world.get_width() > self._rect.width`is not enough (there are some specific case when world size is close to area size)

Use `get_hidden_width() == 0` or `get_hidden_height() == 0` to know if scrollbars are displayed or not

Comment (ppizarror) [reply pygameMenu/scrollarea.py:309]: changed!

**Comment by anxuae** [comment]
> @anxuae. There's an inconsistency in `draw_region_y`.
> 
> After the y scrollbar returns to the first position. In the test I made (see gif) at first the margin is valid (added `draw_region_y=30`), But after returning to the first position the margin was deleted.
> 
> https://i.imgur.com/KXUOohR.gif
> 
> PD: Not deleted, but it should return to the first state drawn

Fixed ;-)

**Comment by ppizarror** [comment]
As menu has more parameters, I renamed `color_selected` to `selection_color` and `rect_width` to `seletion_border_width`. This breaks compatibility with previous versions.

**Comment by ppizarror** [comment]
Also renamed `title_offsetx` and `title_offsety` to `title_offset_x` and `title_offset_y`.
Introduced new parameters `selection_explode_margin_x` and `selection_explode_margin_y`.

**Comment by ppizarror** [comment]
Introduced also a migration guide: https://github.com/ppizarror/pygame-menu/wiki/Migration-Guide

**Comment by anxuae** [comment]
Hi @ppizarror , just a question, how do you add your own commit to my PR?
Can you explain me what git command do you use?
Thanks

**Comment by ppizarror** [comment]
> Hi @ppizarror , just a question, how do you add your own commit to my PR?
> Can you explain me what git commands do you use?
> Thanks

Just use github for windows and vscode. Maybe b/c I have the permissions of `Owner`?

**Comment by ppizarror** [comment]
I think this PR is way too large. I'm gonna merge it, if you want to change anything please create another PR. Thanks!!

**Pull request merged**